packages/tangle/approvalweightmanager.go

```diff
@@ -86,6 +87,70 @@ func (a *ApprovalWeightManager) weightsPerEpoch(branchID ledgerstate.BranchID) *
 	return weightsPerEpoch
 }
 
+func (a *ApprovalWeightManager) onMarkerConfirmed(marker *markers.Marker, messageID MessageID) {
+	// walk through the marker's pastcone to update FinalizedApprovalWeight
+	finalizedWalker := walker.New()
+
+	// push marker itself
+	finalizedWalker.Push(messageID)
```

Comment:
Why do we need to push the message itself if we already do `metadata.SetFinalizedApprovalWeight(true)`?